orttraining/orttraining/core/session/training_session.cc

```diff
@@ -880,6 +891,59 @@ common::Status TrainingSession::GetStateTensors(NameMLValMap& state_tensors) {
   return GetSessionState().GetInitializedTensors(GetStateTensorNames(), allow_missing, state_tensors);
 }
 
+common::Status TrainingSession::GetOptimizerState(std::unordered_map<std::string, NameMLValMap>& opt_state_tensors) {
+  bool allow_missing = (opt_graph_config_.deepspeed_zero.stage != 0);
+  for (const auto& weight_map : weight_to_opt_mapping_) {
+    NameMLValMap curr_opt_tensors;
+    std::unordered_set<std::string> opt_names(weight_map.second.begin(), weight_map.second.end());
+    const auto& weight_name = weight_map.first;
+    GetSessionState().GetInitializedTensors(opt_names, allow_missing, curr_opt_tensors);
+    opt_state_tensors[weight_name] = {};
+    // Keep only prefix in returned value
+    for (const auto& opt_pair: curr_opt_tensors) {
+      const auto& opt_name = opt_pair.first;
+      std::string pre_fix_only = opt_name.substr(0, opt_name.find(weight_name) - 1);
```

Comment:
```const```

Reply:
Added

Reply:
This was removed as of suggest by Aishwarya